### PR TITLE
OpenMPI: Depends on hwlock & libevent

### LINKED ISSUE #20656: Open MPI 4.0.5 build fails on macOS 10.15: libevent
I am trying to install Open MPI 4.0.5 on macOS 10.15 via Spack using AppleClang as the compiler. The instance runs on Azure Pipelines as part of my quality assurance scripts, since I don't own a mac.

Following a build error on macOS 10.15 and checking the homebrew recipes, I think Open MPI depends on libevent.

### Steps to reproduce the issue

```console
$ spack install openmpi
```

### Information on your system

* Operating system/version: macOS 10.15
* Computer hardware: x86_64
* Network type: Azure Pipelines / cloud (single node, 2 vcores)

### Additional information

Build log: [341.txt](https://github.com/spack/spack/files/5765811/341.txt)

Open MPI bug report: https://github.com/open-mpi/ompi/issues/8332

Declared dependencies in Homebrew include `libevent`: https://github.com/Homebrew/homebrew-core/blob/master/Formula/open-mpi.rb

Proposed fix: #20658

### General information

N/A

cc package maintainer @hppritcha

### PR DESCRIPTION
**Pull request by ax3l**
Both hwlock & libevent are required dependencies of Open MPI. While they are also shipped internally, newer releases (>=4.0) will start looking for external packages by default.

This caused build issues of Open MPI 4.0.5 with Fortran on macOS 10.15.

cc @hppritcha

Proposal to fix #20656

Note: I need a macOS user to test this, since I only saw this in my QA scripts on an Azure Cloud instance.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openmpi/package.py
Comment (ax3l) [inline var/spack/repos/builtin/packages/openmpi/package.py]: We can also make this conditional on `@4:` since earlier releases just used the bundled libevent if no external one is specified.
Not sure how many people rely on the very old tagged versions in this package, I did not test the libevent-Open MPI compatibility matrix for those `version`s (libevent in Spack generally ships version 2.x)

**Comment by hppritcha** [comment]
I'd also suggest qualifying the use of external libevent with the OMPI version number, with the preference for external starting at 4.0.0.  The 3.1.x and 3.0.x release streams are still fairly widely used.

**Comment by ax3l** [comment]
Thanks, agreed. PR updated accordingly.

**Comment by tldahlgren** [comment]
Force re-running test.

**Comment by adamjstewart** [comment]
I actually disagree with @hppritcha. It's better to rely on a Spack dependency for all versions instead of a vendored dependency, as Spack packages tend to get more frequent updates and patches for uncommon compilers. But @hppritcha is the maintainer of this package, so I'll defer to his judgement. We can always change it later if we need to.

**Comment by ax3l** [comment]
I generally agree with you @adamjstewart to disagree on this, but I also really don't want to stress the checked compatibility and potential patches that went in older releases of OpenMPI.

And since the problem will be gone with 4.0+ from the Spack perspective, I am ok to keep the vendored version for the oldstable and ancient releases for sake of stability.

**Comment by ax3l** [comment]
@hppritcha @adamjstewart from the :+1: on the last comment - are you ok to merge? :)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
